multinode/bootstrapping.md

```diff
@@ -0,0 +1,201 @@
+node using the [`add_data_node`][add_data_node] and
+[`attach_data_node`][attach_data_node] functions. In order for a data
+node to be available to the access node it is necessary to ensure that
+it is properly setup to work as a data node. This process is referred
+to as *bootstrapping the data node*.
+
+## Connections to the data node
+
+All connections to the data node are made as the current user on the
+access node. Which requires that:
+
+1. The current user exists on both the data node and access node.
+2. The data node is set up to handle logins using either certificates
+   or password files, as described in the [Data Node Authentication
+   Chapter][distdb-auth].
+
+## Adding data nodes to the access node
+
+While an access node is adding a new data node to the distributed
+database it either bootstraps the data node or verifies that
+the data node is properly bootstrapped.
+
+A new data node is added to the access node using the
+[`add_data_node`][add_data_node] command.
+
+```
+SELECT * FROM add_data_node('data_node_1', 'data1.example.com');
+```
+
+The access node has to ensure that the database on the data node is
+properly configured by either bootstrapping the database or verifying
+that the database exists and is correctly configured. This is controlled with the
+`bootstrap` option, which can be either `true` (indicating that the
+data node should be bootstrapped) or `false` (indicating that the data
+node will only be validated, but no bootstrapping will take place).
+
+```
+SELECT * FROM add_data_node('data_node_1', 'data1.example.com',
+                            boostrap => false);
+```
+
+The default value for the `bootstrap` option is `true`, which means
+that the data node is bootstrapped when added.
+
+The steps to boostrap the data node are:
+
+1. Ensure that there is a database on the data node.
+2. Ensure that the Timescale extension is available in the database.
+3. Set metadata on the data node.
+
+## Setting up the database on the data node
+
+The database to bootstrap on the data node is given by the name of the
+current database on the access node. For example, if you are connected
+to the `postgres` database on the access node, the database name will
+be `postgres` on the data node as well.
+
+To provision the data node in a different database on the instance use
+the `database` option to the [`add_data_node`][add_data_node] command,
+for example:
+
+```
+SELECT * FROM add_data_node('data_node_1', 'data1.example.com',
+                            database => 'my_database');
+```
+
+This is particularly useful if you want to experiment with having data
+nodes and access nodes in the same Postgres instance.
+
+### Bootstrapping the database
+
+If `bootstrap` is true (which is the default), the access node will
+first connect to either the `postgres` database or, if the `postgres`
+database does not exist, to the `template0` database on the data node,
+and create a new database:
+
+```
+CREATE DATABASE <database>
```

Comment:
While I think it is great to document things, I am wondering what purpose these detailed bootstrapping docs serve (thinking mostly about stuff from this section and onward)? Do we intend for people to run through these steps manually?

I am worried that this will confuse people more than it helps given that this is automated by the `add_data_node` command. Further, these docs will easily get outdated as soon as we make a small change to the bootstrapping processes. 

If this is about making people better understand what is going on internally, then I think we should rather improve error messages and notices.

Reply:
I've gotten questions about how it works several times when on support. Also, if this procedure changes frequently, then we have done a bad job at the design, so that do not worry me too much. Once we release this, it will be stable only because it has to be stable.

Reply:
Sounds great that we have such interest and it would be interesting to deep-dive into that conversation. I've been following at least the `multinode-beta` channel but haven't seen any questions on the bootstrapping process so far (apart from the Docker extension issue, which is since fixed). Do you have a reference to those questions about the bootstrapping process on support Slack? 

Taking a step back though, the reason I am pushing back a bit is that I generally have the opinion that any need for detailed documentation is a signal that you've failed to provide a good interface. Most people never read the documentation and it easily gets out-of-date, so I am simply trying to manage trade-offs here. I'd rather have engineers spending time on code than docs, instead making sure that the interfaces are clear and that the code works in such a way that it makes documentation almost redundant.

Anyway, if you think this is important I am fine with merging this. I am just trying to look at the bigger docs picture than nitpicking spelling errors, grammar, etc.

Reply:
Yes, I agree that there is a possibility that the documentation will become out-of-date as we move forward, but:

- I do not think it is an argument against providing it both for ourselves and for the users.
- Documenting our intentions will allow users to provide feedback, which will make the product better.
- As we move forward, the documentation will be stable because changes cannot be arbitrarily made to the code and we need to keep these things stable to allow upgrades.